src/common/kube-helpers.ts

```diff
@@ -0,0 +1,162 @@
+export function dumpConfigYaml(kubeConfig: Partial<KubeConfig>): string {
+  const config = {
+    apiVersion: "v1",
+    kind: "Config",
+    preferences: {},
+    'current-context': kubeConfig.currentContext,
+    clusters: kubeConfig.clusters.map(cluster => {
+      return {
+        name: cluster.name,
+        cluster: {
+          'certificate-authority-data': cluster.caData,
+          'certificate-authority': cluster.caFile,
+          server: cluster.server,
+          'insecure-skip-tls-verify': cluster.skipTLSVerify
+        }
+      }
+    }),
+    contexts: kubeConfig.contexts.map(context => {
+      return {
+        name: context.name,
+        context: {
+          cluster: context.cluster,
+          user: context.user,
+          namespace: context.namespace
+        }
+      }
+    }),
+    users: kubeConfig.users.map(user => {
+      return {
+        name: user.name,
+        user: {
+          'client-certificate-data': user.certData,
+          'client-certificate': user.certFile,
+          'client-key-data': user.keyData,
+          'client-key': user.keyFile,
+          'auth-provider': user.authProvider,
+          exec: user.exec,
+          token: user.token,
+          username: user.username,
+          password: user.password
+        }
+      }
+    })
+  }
+
+  logger.debug("Dumping KubeConfig:", config);
+
+  // skipInvalid: true makes dump ignore undefined values
+  return yaml.safeDump(config, { skipInvalid: true });
+}
+
+export function podHasIssues(pod: V1Pod) {
+  // Logic adapted from dashboard
+  const notReady = !!pod.status.conditions.find(condition => {
+    return condition.type == "Ready" && condition.status !== "True"
+  });
+
+  return (
+    notReady ||
+    pod.status.phase !== "Running" ||
+    pod.spec.priority > 500000 // We're interested in high prio pods events regardless of their running status
+  )
+}
+
+// Logic adapted from dashboard
+// see: https://github.com/kontena/kontena-k8s-dashboard/blob/7d8f9cb678cc817a22dd1886c5e79415b212b9bf/client/api/endpoints/nodes.api.ts#L147
+export function getNodeWarningConditions(node: V1Node) {
+  return node.status.conditions.filter(c =>
+    c.status.toLowerCase() === "true" && c.type !== "Ready" && c.type !== "HostUpgrades"
+  )
+}
+
+// Write kubeconfigs to "embedded" store, i.e. "/Users/ixrock/Library/Application Support/Lens/kubeconfigs"
+export function saveConfigToAppFiles(clusterId: string, kubeConfig: KubeConfig | string): string {
+  const userData = (app || remote.app).getPath("userData");
+  const kubeConfigFile = path.join(userData, `kubeconfigs/${clusterId}`)
+  const kubeConfigContents = typeof kubeConfig == "string" ? kubeConfig : dumpConfigYaml(kubeConfig);
+
+  ensureDirSync(path.dirname(kubeConfigFile));
+  writeFileSync(kubeConfigFile, kubeConfigContents);
+  return kubeConfigFile;
+}
+
+export async function getKubeConfigLocal(): Promise<string> {
+  try {
+    const configFile = path.join(process.env.HOME, '.kube', 'config');
```

Comment:
use `os.homedir()`?

Reply:
is it not the same?

Reply:
On Windows it could be `process.env.USERPROFILE`. At least this used to be an issue... 
`os.homedir()` is likely more portable, plus it is used on line 11 above, so we should be consistent.
Sorry, didn't realize that this is code that was only moved, not modified